Analytics/Request/BlockingRequestHandler.cs

```diff
@@ -90,17 +94,19 @@ internal class BlockingRequestHandler : IRequestHandler
         /// </summary>
         public TimeSpan Timeout { get; set; }
 
-        internal BlockingRequestHandler(Client client, TimeSpan timeout) : this(client, timeout, null)
+        internal BlockingRequestHandler(Client client, TimeSpan timeout) : this(client, timeout, null, new Backo(max: 10000, jitter: 5000)) // Set maximum waiting limit to 10s and jitter to 5s
         {
         }
 #if NET35
 
-        internal BlockingRequestHandler(Client client, TimeSpan timeout, IHttpClient httpClient)
+        internal BlockingRequestHandler(Client client, TimeSpan timeout, IHttpClient httpClient, Backo backo)
 #else
-        internal BlockingRequestHandler(Client client, TimeSpan timeout, HttpClient httpClient)
+        internal BlockingRequestHandler(Client client, TimeSpan timeout, HttpClient httpClient, Backo backo)
 #endif
         {
             this._client = client;
+            _backo = backo;
```

Comment:
What happens if `backo` isn't defined? Are we requiring `backo` now in the interface? Or can it be optional still?

Reply:
if backo isn't defined an exception is thrown. Backo is required because it implements the retry strategy, but these constructors are internal, this means that the final user doesn't have access to them. The client class uses the constructor with a default instantiation of backo (10s max and 5s of jitter), the other constructor was added for testing purposes.